internal/services/springcloud/spring_cloud_service_resource.go

```diff
@@ -651,9 +662,16 @@ func expandSpringCloudTrace(input []interface{}) *appplatform.MonitoringSettingP
 		}
 	}
 	v := input[0].(map[string]interface{})
+	instrumentationKey := ""
+	if value := v["instrumentation_key"]; len(value.(string)) > 0 {
+		instrumentationKey = value.(string)
+	}
+	if value := v["connection_string"]; len(value.(string)) > 0 {
+		instrumentationKey = value.(string)
+	}
 	return &appplatform.MonitoringSettingProperties{
 		TraceEnabled:                  utils.Bool(true),
-		AppInsightsInstrumentationKey: utils.String(v["instrumentation_key"].(string)),
+		AppInsightsInstrumentationKey: utils.String(instrumentationKey),
```

Comment:
is there no new property on the backend for this/is it forthcoming in a new API version?

Reply:
No, there isn't.